src/api/libcellml/analysermodel.h

```diff
@@ -0,0 +1,452 @@
+     *
+     * @return The equations.
+     */
+    std::vector<AnalyserEquationPtr> equations() const;
+
+    /**
+     * @brief Get the equation at @p index.
+     *
+     * Return the equation at the index @p index for the @c Model processed by
+     * this @c Analyser.
+     *
+     * @param index The index of the equation to return.
+     */
+    AnalyserEquationPtr equation(size_t index) const;
+
+    /**
+     * @brief Test to determine if @c AnalyserModel needs the equality function.
+     *
+     * Test to determine if @c AnalyserModel needs the equality function, return
+     * @c true if it does and @c false otherwise.
+     *
+     * @return @c true if @c AnalyserModel needs the equality function, @c false
+     * otherwise.
+     */
+    bool needEqFunction() const;
+
+    /**
+     * @brief Test to determine if @c AnalyserModel needs the inequality
+     * function.
+     *
+     * Test to determine if @c AnalyserModel needs the inequality function,
+     * return @c true if it does and @c false otherwise.
+     *
+     * @return @c true if @c AnalyserModel needs the inequality function,
+     * @c false otherwise.
+     */
+    bool needNeqFunction() const;
+
+    /**
+     * @brief Test to determine if @c AnalyserModel needs the lower than
+     * function.
+     *
+     * Test to determine if @c AnalyserModel needs the lower than function,
+     * return @c true if it does and @c false otherwise.
+     *
+     * @return @c true if @c AnalyserModel needs the lower than function,
+     * @c false otherwise.
+     */
+    bool needLtFunction() const;
+
+    /**
+     * @brief Test to determine if @c AnalyserModel needs the lower or equal
+     * than function.
+     *
+     * Test to determine if @c AnalyserModel needs the lower or equal than
+     * function, return @c true if it does and @c false otherwise.
+     *
+     * @return @c true if @c AnalyserModel needs the lower or equal than
+     * function, @c false otherwise.
+     */
+    bool needLeqFunction() const;
+
+    /**
+     * @brief Test to determine if @c AnalyserModel needs the greater than
+     * function.
+     *
+     * Test to determine if @c AnalyserModel needs the greater than function,
+     * return @c true if it does and @c false otherwise.
+     *
+     * @return @c true if @c AnalyserModel needs the greater than function,
+     * @c false otherwise.
+     */
+    bool needGtFunction() const;
+
+    /**
+     * @brief Test to determine if @c AnalyserModel needs the greater or equal
+     * than function.
+     *
+     * Test to determine if @c AnalyserModel needs the greater or equal than
+     * function, return @c true if it does and @c false otherwise.
+     *
+     * @return @c true if @c AnalyserModel needs the greater or equal than
+     * function, @c false otherwise.
+     */
+    bool needGeqFunction() const;
+
+    /**
+     * @brief Test to determine if @c AnalyserModel needs the and function.
```

Comment:
How's this for nit-picky ... should the "and" be in all caps? the AND function? or the "and" function? Or something else that makes this easier to read?

Reply:
Also below.

Reply:
Agreed, it doesn't look great. Ok, will go with "and" since we can use this  for all types of functions. Fixed in commit 4762f45. (Why did I agree to this?! Took me forever!)